src/pages/orderpage/OrderPage.tsx

```diff
@@ -0,0 +1,103 @@
+/** @jsxImportSource @emotion/react */
+import styled from "@emotion/styled";
+import { useEffect } from "react";
+import { useNavigate, useParams } from "react-router-dom";
+import MessageCardSection from "@/pages/orderpage/MessageCardSection";
+import SenderInfoSection from "@/pages/orderpage/SenderInfoSection";
+import ReceiverInfoSection from "@/pages/orderpage/RecieverSection";
+import ProductSummarySection from "@/pages/orderpage/ProductSummarySection";
+import { useForm } from "@/hooks/useForm";
+import OrderButton from "@/components/common/BaseButton";
+import { MOCK_PRODUCTS } from "@/mocks/products_list_mock";
+import {
+  validateMessage,
+  validateSender,
+  validateReceiver,
+  validatePhone,
+  validateQuantity,
+} from "@/utils/validator";
+
+const OrderPage = () => {
+  const navigate = useNavigate();
+  const { id } = useParams();
+  const product = MOCK_PRODUCTS.find((item) => item.id === Number(id));
+
+  const { values, setValues, errors, validateAll } = useForm(
+    {
+      message: "",
+      sender: "",
+      receiver: "",
+      phone: "",
+      quantity: 1,
+    },
+    {
+      message: validateMessage,
+      sender: validateSender,
+      receiver: validateReceiver,
+      phone: validatePhone,
+      quantity: validateQuantity,
+    }
+  );
+
+  useEffect(() => {
+    if (!product) {
+      navigate("/notfound", { replace: true });
+    }
+  }, [product]);
+
+  if (!product) return null;
```

Comment:
여기도 Navigate 컴포넌트를 리턴하는 방식을 사용할 수 있겠네요

```suggestion
  if (!product) {
    return <Navigate to="/notfound" replace />
  }
```